server/core/src/main/java/io/whitefox/core/services/AwsGlueConfigBuilder.java

```diff
@@ -0,0 +1,40 @@
+package io.whitefox.core.services;
+
+import io.whitefox.core.AwsCredentials;
+import io.whitefox.core.MetastoreProperties;
+import io.whitefox.core.aws.utils.StaticCredentialsProvider;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.iceberg.CatalogProperties;
+import org.apache.iceberg.aws.AwsClientProperties;
+import org.apache.iceberg.aws.AwsProperties;
+
+public class AwsGlueConfigBuilder {
+
+  public Map<String, String> buildConfig(
+      MetastoreProperties.GlueMetastoreProperties glueMetastoreProperties) {
+    try {
```

Comment:
This is not the way to check that a cast is safe.
You should do something along the lines of:

```
if (glueMetastoreProperties instanceOf AwsCredentials.SimpleAwsCredentials) {
  AwsCredentials.SimpleAwsCredentials credentials =
          (AwsCredentials.SimpleAwsCredentials) glueMetastoreProperties.credentials();
  ....
} else {
  throw new IllegalArgumentException("Credentials type not supported with glue metastore %s")
}
```

Reply:
fixed